contracts/SynthetixRouterStrategy.sol

```diff
@@ -0,0 +1,351 @@
+    }
+
+    function checkWaitingPeriod() private returns (bool freeToMove) {
+        return
+            _exchanger().maxSecsLeftInWaitingPeriod(
+                address(this),
+                synthCurrencyKey
+            ) == 0;
+    }
+
+    function depositInVault() public onlyVaultManagers {
+        uint256 balanceOfSynth = _balanceOfSynth();
+        if (balanceOfSynth > DUST_THRESHOLD && checkWaitingPeriod()) {
+            _checkAllowance(
+                address(yVault),
+                address(_synthCoin()),
+                balanceOfSynth
+            );
+            yVault.deposit();
+        }
+    }
+
+    //safe to enter more than we have
+    function withdrawSomeWant(uint256 _amount, bool performExchanges)
+        private
+        returns (uint256 _liquidatedAmount, uint256 _loss)
+    {
+        if (performExchanges) {
+            // we exchange synths to susd
+            uint256 synthBalanceBefore = _balanceOfSynth();
+            uint256 sUSDBalanceBefore = balanceOfWant();
+            uint256 _new_amount = _amount.sub(sUSDBalanceBefore);
+
+            uint256 _synth_amount = _synthFromSUSD(_new_amount);
+            uint256 _remaining_amount = _synth_amount;
+            if (checkWaitingPeriod()) {
+                if (_synth_amount <= synthBalanceBefore) {
+                    exchangeSynthToSUSD(_synth_amount);
+                    return (_amount, 0);
+                }
+
+                _remaining_amount = _synth_amount.sub(synthBalanceBefore);
+            }
+            _withdrawFromYVault(_remaining_amount);
+            uint256 newBalanceOfSynth = _balanceOfSynth();
+            if (newBalanceOfSynth > DUST_THRESHOLD) {
+                exchangeSynthToSUSD(newBalanceOfSynth);
+            }
+        }
+
+        uint256 totalAssets = balanceOfWant();
+        if (_amount > totalAssets) {
+            _liquidatedAmount = totalAssets;
+            _loss = _amount.sub(totalAssets);
+        } else {
+            _liquidatedAmount = _amount;
+        }
+    }
+
+    function liquidateAllPositions()
+        internal
+        override
+        returns (uint256 _amountFreed)
+    {
+        // In order to work, manualRemoveFullLiquidity needs to be call 6 min in advance
+        require(checkWaitingPeriod(), "Wait for settlement period");
+        require(
+            valueOfInvestment() == 0,
+            "Need to remove liquidity from vault first"
+        );
+        require(_balanceOfSynth() == 0, "Need to exchange synth to want first");
+        _amountFreed = balanceOfWant();
+    }
+
+    function manualRemoveFullLiquidity()
+        external
+        onlyVaultManagers
+        returns (uint256 _liquidatedAmount, uint256 _loss)
+    {
+        // It will remove max amount of assets and trade synth for sUSD
+        // the Synthetix waiting period will start (and harvest can be called 6 mins later)
+        if (yVault.balanceOf(address(this)) > 0) {
+            super.liquidateAllPositions();
```

Comment:
does it make sense to replace this with the withdrawing from the vault? to improve readability

Reply:
I tried avoiding repeating code from the parent class. Almost all yvault logic is there. I understand that it will improve readability but why having classes and inheritance then?